x-pack/plugins/task_manager/server/task_running/errors.ts

```diff
@@ -14,9 +14,17 @@ const CODE_SKIP = 'TaskManager/skip';
 const code = Symbol('TaskManagerErrorCode');
 const retry = Symbol('TaskManagerErrorRetry');
 
-export interface DecoratedError extends Error {
+export enum TaskErrorSource {
+  FRAMEWORK = 'framework',
+  USER = 'user',
+  RULE_TYPE = 'rule_type',
+  CONNECTOR_TYPE = 'connector_type',
+}
+
+export interface TaskRunError extends Error {
```

Comment:
Renamed DecoratedError as TaskRunError and added the source field.

Reply:
I don't really understand why `RULE_TYPE` and `CONNECTOR_TYPE` are here ... the two cases I saw of `RULE_TYPE` are I think "user" error.